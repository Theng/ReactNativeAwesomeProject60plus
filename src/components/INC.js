import React from "react";
import { Button } from "react-native";
function MyComponent({increment,label}) {
    console.log("Render increase button Compoenent.");
    return (
        <Button
            title={label}
            onPress={()=>increment(5)}
        />
    );
}

export default React.memo(MyComponent);
