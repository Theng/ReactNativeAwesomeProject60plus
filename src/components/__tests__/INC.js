import React from 'react';
import INC from '../INC';
import {render, cleanup} from 'react-native-testing-library';

afterEach(cleanup);

describe('<INC />', () => {
  it('should match snapshot', () => {
    const rendered = render(<INC label="increase"/>).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});