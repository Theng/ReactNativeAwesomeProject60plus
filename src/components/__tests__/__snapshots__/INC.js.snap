// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<INC /> should match snapshot 1`] = `
<View
  accessibilityRole="button"
  accessibilityStates={Array []}
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "padding": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      increase
    </Text>
  </View>
</View>
`;
